src/Microsoft.ApplicationInsights.AspNetCore/Extensions/ApplicationInsightsExtensions.cs

```diff
@@ -129,8 +131,8 @@ public static IServiceCollection AddApplicationInsightsTelemetry(this IServiceCo
             {
                 services.TryAddSingleton<IHttpContextAccessor, HttpContextAccessor>();
 
-                services.AddSingleton<ITelemetryInitializer, AzureWebAppRoleEnvironmentTelemetryInitializer>();
-                services.AddSingleton<ITelemetryInitializer, DomainNameRoleInstanceTelemetryInitializer>();
+                services.AddSingleton<ITelemetryInitializer, ApplicationInsights.AspNetCore.TelemetryInitializers.AzureWebAppRoleEnvironmentTelemetryInitializer>();
```

Comment:
Can this be taken from the `WindowsServer` package? If not - aspnetcore and web might set different values for role name and role instance

Reply:
It most certainly can. I've added issue #629 to track this as a separate issue. There will need to be some work to consolidate the modules properly as the implementations have diverged.